zebra-rpc/src/methods.rs

```diff
@@ -645,6 +645,64 @@ where
             })
     }
 
+    #[cfg(feature = "getblocktemplate-rpcs")]
+    // TODO: use a generic error constructor (#5548)
+    fn get_raw_mempool(&self) -> BoxFuture<Result<Vec<String>>> {
+        use std::cmp::Ordering;
+        use zebra_chain::transaction::VerifiedUnminedTx;
+
+        /// Returns relative score for sorting [`VerifiedUnminedTx`]s (fee/size)
+        fn tx_sort_score(tx: &VerifiedUnminedTx, tx2: &VerifiedUnminedTx) -> Option<u64> {
```

Comment:
This seems like a direct translation of the C code into Rust, but it's not clear to me exactly how it is achieving the desired order. (Or if the code is doing it correctly.)

There are also more efficient ways to do the same sort in Rust, so our code can be simpler than the C version.

In Rust, one idiomatic and efficient way to do this is:
https://doc.rust-lang.org/std/primitive.slice.html#method.sort_by_cached_key

So we might be able to use something like:
```rust
if USE_ZCASHD_ORDER {
    transactions.sort_by_cached_key(|tx| {
        Reverse((
            tx.miner_fee as f32 / tx.transaction.size as f32,
            tx.transaction.id.mined_id().bytes_in_display_order(),
        ))
    });
}
```

Or maybe just with the Reverse around fee/size, if the hash is sorted in ascending order.

Reply:
We could also use f64 if you wanted to match the zcashd code exactly. The performance difference would be minimal.

Reply:
Floats don't implement `Ord` which is a constraint for `sort_by_key` and `sort_by_cached_key`, it could use `sort_by` with `partial_cmp` or u64 division if sorted by key.

Reply:
Updated in [`2d7cf50`](https://github.com/ZcashFoundation/zebra/pull/5994/commits/2d7cf5004d277ed2cfbb16d6be58767d79b7e1cb) to use `fee * MAX_BLOCK_BYTES / tx.size` with a `u128`